src/game/AI/ScriptDevAI/scripts/outland/hellfire_citadel/shattered_halls/shattered_hallsScripts.cpp

```diff
@@ -0,0 +1,279 @@
+        }
+        else if (eventType == AI_EVENT_CUSTOM_B) // End Duel
+        {
+            m_creature->AttackStop();
+            m_creature->ClearTemporaryFaction();
+            m_creature->HandleEmote(EMOTE_ONESHOT_BOW);
+            ResetTimer(GLADIATIOR_EVENT_RP, 1000);
+            m_eventStarted = false;
+        }
+        else if (eventType == AI_EVENT_CUSTOM_C) // End Duel
+        {
+            m_creature->SetFactionTemporary(1692, TEMPFACTION_RESTORE_RESPAWN | TEMPFACTION_RESTORE_COMBAT_STOP);
+            AttackStart(invoker);
+            m_sparringPartner = invoker->GetObjectGuid();
+        }
+    }
+    void ExecuteAction(uint32 action) override
+    {
+        switch (action)
+        {
+        case GLADIATOR_STOP_EVENT:
+        {
+            if (m_creature->GetHealthPercent() <= 40.f)
+            {
+                if (Creature* centurion = GetClosestCreatureWithEntry(m_creature, NPC_SHATTERED_HAND_CENTURION, 15.f, true, false, true))
+                    SendAIEvent(AI_EVENT_CUSTOM_A, m_creature, centurion);
+            }
+            break;
+        }
+        }
+    }
+    void UpdateAI(const uint32 diff) override
+    {
+        CombatAI::UpdateAI(diff);
+        if (m_creature->IsInCombat() && m_eventStarted)
+        {
+            if (m_creature->SelectAttackingTarget(ATTACKING_TARGET_TOPAGGRO, 0, nullptr, SELECT_FLAG_PLAYER))
+            {
+                if (Creature* partner = m_creature->GetMap()->GetCreature(m_sparringPartner))
+                {
+                    SendAIEvent(AI_EVENT_CUSTOM_A, m_creature, partner);
+                    m_creature->getThreatManager().modifyThreatPercent(partner, -101);
+                    m_creature->ClearTemporaryFaction();
+                    m_eventStarted = false;
+                }
+            }
+        }
+    }
+};
+
+struct npc_shattered_hand_centurion : public CombatAI
+{
+    npc_shattered_hand_centurion(Creature* creature) : CombatAI(creature, GLADIATOR_ACTION_MAX),
+        m_instance(static_cast<ScriptedInstance*>(creature->GetInstanceData())),
+        m_eventStarted(false)
+    {
+        AddCustomAction(GLADIATOR_START_EVENT, 4000u, [&]() { HandleEventStart(); });
+    }
+
+    void Reset() override
+    {
+        CombatAI::Reset();
+        m_eventStarted = false;
+    }
+
+    ScriptedInstance* m_instance;
+    bool m_eventStarted;
+
+    void Aggro(Unit* /*who*/) override
+    {
+        switch (urand(0, 5))
+        {
+        case 0: DoBroadcastText(SAY_AGGRO1, m_creature); break;
+        case 1: DoBroadcastText(SAY_AGGRO2, m_creature); break;
+        case 2: DoBroadcastText(SAY_AGGRO3, m_creature); break;
+        case 3: DoBroadcastText(SAY_AGGRO4, m_creature); break;
+        case 4: DoBroadcastText(SAY_AGGRO5, m_creature); break;
+        case 5: DoBroadcastText(SAY_AGGRO6, m_creature); break;
+        case 6: DoBroadcastText(SAY_AGGRO7, m_creature); break;
+        }
+
+        // Set Gladiators infight too and reset their factions
```

Comment:
When using SpawnGroup flag: CREATURE_GROUP_AGGRO_TOGETHER, the Centurion staying in the pit with the Gladiators will also try to attack one of the Gladiators.

4 Gladiators + 1 Centurion = 1 spawn_group with flag CREATURE_GROUP_RESPAWN_TOGETHER

Reply:
Yeah, currently there is no mechanism to bypass that. I strongly suspect blizzard doesnt have flags like this and is forced to script it each time they want to add such behaviour.

To get this straight, during normal player aggro they are meant to aggro together, but during the scripted action it should be disabled?

Reply:
When a player attacks the group, they should always attack together doesnt matter if they are fighting each other or not.

Reply:
https://youtu.be/XvFvPXcR54w 

This is what happens with CREATURE_GROUP_AGGRO_TOGETHER flag set in spawn_group.

Reply:
Simple fix that comes to mind: I will disable aggroing members of same group aggroing each other through CREATURE_GROUP_AGGRO_TOGETHER. Going forward, I will try to design an API where you can call the aggro yourself in a C++ AI in case we need more control. Like, CreatureAI->Aggro IF enemy is player, go full spawn group aggro, if specific npc, ignore that.

Reply:
Yeah works with the flag now, removed the hackfixed code

Reply:
> I will disable aggroing members of same group aggroing each other through CREATURE_GROUP_AGGRO_TOGETHER.

Just in case it's literally going to be what you say and wouldn't be covered, here's a random thought: I'm a priest and mind control one. They should all aggro, and also attack the mind controlled one.

Reply:
if ControlledByPlayer() aggro; I'm guessing that would be the common case

Reply:
i mean if you aggro stuff with your pet currently stuff will directly go for player so there is more wrong then just MC case.

Reply:
> > I will disable aggroing members of same group aggroing each other through CREATURE_GROUP_AGGRO_TOGETHER.
> 
> Just in case it's literally going to be what you say and wouldn't be covered, here's a random thought: I'm a priest and mind control one. They should all aggro, and also attack the mind controlled one.

*sigh